addons/payment/models/payment_acquirer.py

```diff
@@ -53,105 +52,87 @@ class PaymentAcquirer(osv.Model):
-    }
-
-    image = openerp.fields.Binary("Image", attachment=True,
-        help="This field holds the image used for this provider, limited to 1024x1024px")
-    image_medium = openerp.fields.Binary("Medium-sized image",
-        compute='_compute_images', inverse='_inverse_image_medium', store=True, attachment=True,
+    _provider_selection = lambda self: self._get_providers()
+
+    name = fields.Char(required=True, translate=True)
+    provider = fields.Selection(_provider_selection, required=True)
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.user.company_id, required=True)
+    pre_msg = fields.Html(string='Help Message', translate=True, help='Message displayed to explain and help the payment process.')
+    post_msg = fields.Html(string='Thanks Message', help='Message displayed after having done the payment process.')
+    view_template_id = fields.Many2one('ir.ui.view', string='Form Button Template', required=True)
+    registration_view_template_id = fields.Many2one('ir.ui.view', string='S2S Form Template', domain=[('type', '=', 'qweb')], help="Template for method registration")
+    environment = fields.Selection([('test', 'Test'), ('prod', 'Production')], oldname='env', default='prod')
+    website_published = fields.Boolean(string='Visible in Portal / Website', copy=False, default=False, help="Make this payment acquirer available (Customer invoices, etc.)")
+    auto_confirm = fields.Selection(
+        [('none', 'No automatic confirmation'),
+         ('at_pay_confirm', 'At payment with acquirer confirmation'),
+         ('at_pay_now', 'At payment no acquirer confirmation needed')],
+        string='Order Confirmation', default='at_pay_confirm', required=True)
+    pending_msg = fields.Html(string='Pending Message', translate=True,
+        default="<i>Pending,</i> Your online payment has been successfully processed. But your order is not validated yet.", help='Message displayed, if order is in pending state after having done the payment process.')
+    done_msg = fields.Html(string='Done Message', translate=True,
+        default="<i>Done,</i> Your online payment has been successfully processed. Thank you for your order.", help='Message displayed, if order is done successfully after having done the payment process.')
+    cancel_msg = fields.Html(string='Cancel Message', translate=True,
+        default="<i>Cancel,</i> Your payment has been cancelled.", help='Message displayed, if order is cancel during the payment process.')
+    error_msg = fields.Html(string='Error Message', translate=True,
+        default="<i>Error,</i> Please be aware that an error occurred during the transaction. The order has been confirmed but won't be paid. Don't hesitate to contact us if you have any questions on the status of your order.", help='Message displayed, if error is occur during the payment process.')
+    # Fees
+    fees_active = fields.Boolean(string='Add Extra Fees')
```

Comment:
If i am wrong correct me but @gpa-odoo  are you kidding with code :question: @dbu-odoo  :+1: